### PR TITLE
Move fuser to test_jit_fuser

### PR DESCRIPTION
**Pull request by eellison**
Start of breaking up test_jit.py  

New files will have the format test_jit_* so they are easily grepable but remain in the same directory so we don't have to go through multiple sources for imports. 

I am adding a test that's expected to fail to be sure it's running.

### CODE REVIEW

**Review by wanchaol — COMMENTED** [review]
Can you sync with @ailzhang on the merging this together with #18509? It looks like the changes gonna conflict a lot.

File: test_jit_fuser.py
Comment (ailzhang) [inline test_jit_fuser.py]: A lot of these imports are not actually used in this file. Good to remove them. ;) 

I think this PR doesn't conflict with my change ( hopefully :P ) since my PR was mainly touching nn_functional_tests and autograd_method_tests which are not used here.  Cheers!

**Comment by eellison** [comment]
> Can you sync with @ailzhang on the merging this together with #18509? It looks like the changes gonna conflict a lot.

@ailzhang there is only one conflict. test_chunk_constant_script_ad. I can either take that test and move it into this patch or you can rebase after this is landed. Since this is an easy change and it looks like yours needs a fair amount of reviewing i would rather land this first. What do you think?

**Comment by ailzhang** [comment]
@eellison   Yea `test_chunk_constant_script_ad` is easy change, I'll rebase after this lands. Thanks!

File: test_jit_fuser.py
Comment (eellison) [reply test_jit_fuser.py]: thank!

**Review by ailzhang — APPROVED** [review]
LGTM if CI passes! Thanks!

File: test/test_jit_fuser.py
Comment (wanchaol) [inline test/test_jit_fuser.py]: surprised we don't need to add those things in test_jit..

**Review by wanchaol — COMMENTED** [review]
I think we were about to make a `test/jit` folder instead of making a separate file, did you run into any issue when doing that?

**Comment by eellison** [comment]
> I think we were about to make a `test/jit` folder instead of making a separate file, did you run into any issue when doing that?

**Comment by eellison** [comment]
> I think we were about to make a `test/jit` folder instead of making a separate file, did you run into any issue when doing that?

It means you don't need an extra level of indirection to import something + i didn't have to chance the way everyone invokes tests + easier

**Comment by apaszke** [comment]
Acutally why? Can't we just keep separate JIT components in separate test cases?

**Comment by eellison** [comment]
@apaszke if the file gets too large it takes too long to run lint on it, and git 404s when you try to look at its history because it cant deal with the size. I've also been unable to generate code coverage because our internal test system fails with this size as well. Have reached out to the test infra people multiple times but havent gotten any movement.

We can also move it to test/jit/, but then you cant do relative importing and need a pytorch_common_utils extra step. i think whatever we do will be kind of annoying

**Comment by suo** [comment]
@eellison it seems these tests are no longer being run when we run test_jit.py. I want to preserve the idea that we can run all "jit related" tests with a single command.

Can we just put all component tests in a `jit/` folder and import them into test_jit.py? That way we preserve a single point of entry for all JIT tests but have better organization.

**Comment by eellison** [comment]
Yep, i will do that next. Hard to find a time to land moving test_jit but that’s next up

**Pull request closed without merging**
